include/cvc5/c/cvc5.h

```diff
@@ -0,0 +1,3936 @@
+
+/**
+ * Determine if given sort is a Set sort.
+ * @param sort The sort.
+ * @return True if the sort is a Set sort.
+ */
+bool cvc5_sort_is_set(Cvc5Sort sort);
+
+/**
+ * Determine if given sort is a Bag sort.
+ * @param sort The sort.
+ * @return True if the sort is a Bag sort.
+ */
+bool cvc5_sort_is_bag(Cvc5Sort sort);
+
+/**
+ * Determine if given sort is a Sequence sort.
+ * @param sort The sort.
+ * @return True if the sort is a Sequence sort.
+ */
+bool cvc5_sort_is_sequence(Cvc5Sort sort);
+
+/**
+ * Determine if given sort is an abstract sort.
+ * @param sort The sort.
+ * @return True if the sort is a abstract sort.
+ *
+ * @warning This function is experimental and may change in future versions.
+ */
+bool cvc5_sort_is_abstract(Cvc5Sort sort);
+
+/**
+ * Determine if given sort is an uninterpreted sort.
+ * @param sort The sort.
+ * @return True if given sort is an uninterpreted sort.
+ */
+bool cvc5_sort_is_uninterpreted_sort(Cvc5Sort sort);
+
+/**
+ * Determine if given sort is an uninterpreted sort constructor.
+ *
+ * An uninterpreted sort constructor has arity > 0 and can be instantiated to
+ * construct uninterpreted sorts with given sort parameters.
+ *
+ * @param sort The sort.
+ * @return True if given sort is of sort constructor kind.
+ */
+bool cvc5_sort_is_uninterpreted_sort_constructor(Cvc5Sort sort);
+
+/**
+ * Determine if given sort is an instantiated (parametric datatype or
+ * uninterpreted sort constructor) sort.
+ *
+ * An instantiated sort is a sort that has been constructed from
+ * instantiating a sort with sort arguments (see cvc5_sort_instantiate().
+ *
+ * @param sort The sort.
+ * @return True if given sort is an instantiated sort.
+ */
+bool cvc5_sort_is_instantiated(Cvc5Sort sort);
+
+/**
+ * Get the associated uninterpreted sort constructor of an instantiated
+ * uninterpreted sort.
+ *
+ * @param sort The sort.
+ * @return The uninterpreted sort constructor sort.
+ */
+Cvc5Sort cvc5_sort_get_uninterpreted_sort_constructor(Cvc5Sort sort);
+
+/**
+ * Get the underlying datatype of a datatype sort.
+ * @param sort The sort.
+ * @return The underlying datatype of a datatype sort.
+ */
+Cvc5Datatype* cvc5_sort_get_datatype(Cvc5Sort sort);
+
+/**
+ * Instantiate a parameterized datatype sort or uninterpreted sort
+ * constructor sort.
+ *
+ * Create sort parameters with cvc5_mk_param_sort().
+ *
+ * @param sort The sort to instantiate.
+ * @param params The list of sort parameters to instantiate with.
+ * @return The instantiated sort.
+ */
+Cvc5Sort cvc5_sort_instantiate(Cvc5Sort sort, Cvc5Sort params[]);
+
+/**
+ * Get the sorts used to instantiate the sort parameters of a given parametric
+ * sort (parametric datatype or uninterpreted sort constructor sort,
+ * see cvc5_sort_instantiate();
+ *
+ * @param sort The sort.
+ * @return The sorts used to instantiate the sort parameters of a
+ *         parametric sort
+ */
+Cvc5Sort* cvc5_sort_get_instantiated_parameters(Cvc5Sort sort);
```

Comment:
Here (and in other cases below), I guess you're returning an array?  What it the size of the array?  This should be documented.

Reply:
Good catch, this is missing a `size_t* size` output parameter (similar as in other functions that return arrays).

Reply:
Addressed.